content/blog/coreboot-on-talos2.md

```diff
@@ -0,0 +1,273 @@
+
+Steps for flashing Heads are similar to those done for coreboot.
+
+1. Copy the Heads binary to the BMC (assuming in the Heads root directory):
+
+    ```
+    scp -O build/zImage.bundled root@<BMC_IP>:/tmp/zImage.bundled
+    ```
+
+2. Log in to the BMC:
+
+    ```
+    ssh root@<BMC_IP>
+    ```
+
+3. Flash the BOOTKERNEL partition with Heads:
+
+    ```
+    pflash -e -P BOOTKERNEL -p /tmp/zImage.bundled
+    ```
+
+Answer yes to the prompt and wait for the process to finish. After that, start
+the platform and begin [configuring Heads](https://osresearch.net/Configuring-Keys/).
+
+## PNOR emulation
+
+Flash device can be emulated by BMC, which is something we were often using for
+development and testing. This saves a lot of time which would be spent flashing,
+as well as reduces the wear of flash device.
+
+However, this still requires System Package v2.00, and if this is different than
+what real flash holds, SEEPROM will have to be updated when switching between
+physical and emulated image. Also, this approach doesn't survive BMC reboots and
+power losses. BMC doesn't have enough space to keep full PNOR image in
+non-volatile memory, so `tmpfs` must be used for emulation. Don't try to put
+more than one image in `tmpfs` or BMC **will** run out of RAM, which most likely
+will require manual power cycle.
+
+To start, obtain full flash image, either by downloading it from
+[RaptorCS release page](https://wiki.raptorcs.com/wiki/Talos_II/Firmware) or
+reading from existing image on BMC with:
+
+```
+pflash -r /tmp/talos.pnor
+```
+
+After that, you can "flash" the partitions mentioned earlier by adding
+additional parameters to use the file instead of physical flash:
+
+```
+pflash -f -P <partition> -p <image_file> -F /tmp/talos.pnor
+```
+
+Change `<partition>` to one of `HBB`, `HBI`, `BOOTKERNEL` and `<image_file>` to
+`/tmp/bootblock.signed.ecc`, `/tmp/coreboot.rom.signed.ecc` or
+`/tmp/zImage.bundled`, respectively.
+
+To tell BMC to present the contents of this file as flash, run:
+
+```
+mboxctl --backend file:/tmp/talos.pnor
+```
+
+Sometimes this command fails with a timeout, in that case run it again until it
+succeeds.
+
+> We've noticed that sometimes, despite no error message printed, physical flash
+> was used anyway. It is easy to spot when one copy has Hostboot and the other
+> has coreboot, but it can be missed when both images have different versions of
+> coreboot. It caused us few hours of unnecessary debugging of issues that were
+> already fixed...
+
+With the file now mounted, platform can be started. Host firmware and OS
+shouldn't be able to tell the difference, except for different reported erase
+block size and maybe different access times.
+
+To get back to original flash, run:
+
+```
+mboxctl --backend vpnor
+```
+
+It will report an error (`Failed to post message: Connection timed out`), but
+will revert to physical device nonetheless. This can be confirmed by checking
+the output of `mboxctl --lpc-state`.
```

Comment:
An output of the state check command would be helpful.

Reply:
Added: https://github.com/3mdeb/openpower-foundation-website/compare/89f864e1cf8dbe6095e01840dac222d2e73256de..cdb1949a6559800e6bb13cbea70060348116da21#diff-41a3cc642dd6f4e4418726ab8e4f9333c28bd09ad4b8d13ebb5bf8417d31a098R230